### PR TITLE
Do not update status bar for 'widgets'

### PR DESCRIPTION
**Pull request by kaste**
Sublime widgets are e.g. the command palette, and the input box at the bottom.
For these views we generally do not need to update the status bar.

Views without a window are also now called transient. These are detached views
after close.

### CODE REVIEW

**Comment by kaste** [comment]
If you enable logging, you can clearly see that we call `rev-parse toplevel` 5(!) times per `ctrl+p`.

**Comment by randy3k** [comment]
It makes sense to me

**Comment by asfaltboy** [comment]
Just gotta say, @kaste I ❤️ that you keep improving performance; bravo!

File: core/commands/status_bar.py [hunk 1]
```diff
@@ -27,19 +27,29 @@ def on_post_save(self, view):
 update_status_bar_soon = False
 
 
+def view_is_transient(view):
```
Comment (asfaltboy) [inline core/commands/status_bar.py]: The description of this PR can go into this function's docstring

Comment (kaste) [reply core/commands/status_bar.py]: 🙄 	I skipped reading the mail after seeing the ❤️ 	above.

Comment (asfaltboy) [reply core/commands/status_bar.py]: Only noticed that this could be improved after sending the approval comment

**Comment by kaste** [comment]
Here we also have a too many bugs case: The throttle function is not good enough. We overcall for new views and loading views. Also: for activating unnamed views we call `git toplevel` 5 times bc the repo path isn't cached.

**Comment by kaste** [comment]
Another bug: For secondary views (clones) the status bar will not update after save.

**Comment by randy3k** [comment]
Let's merge this first and address the bugs later.

**Comment by randy3k** [comment]
another bug found: #1099

**Pull request merged**